temporal-workflowcheck/src/main/java/io/temporal/workflowcheck/Main.java

```diff
@@ -0,0 +1,143 @@
+
+/** Entrypoint for CLI. */
+public class Main {
+  public static void main(String[] args) throws IOException {
+    if (args.length == 0 || "--help".equals(args[0])) {
+      System.err.println(
+          "Analyze Temporal workflows for common mistakes.\n"
+              + "\n"
+              + "Usage:\n"
+              + "  workflowcheck [command]\n"
+              + "\n"
+              + "Commands:\n"
+              + "  check - Check all workflow code on the classpath for invalid calls\n"
+              + "  prebuild-config - Pre-build a config for certain packages to keep from scanning each time (TODO)");
+      return;
+    }
+    switch (args[0]) {
+      case "check":
+        System.exit(check(Arrays.copyOfRange(args, 1, args.length)));
+      case "prebuild-config":
+        System.exit(prebuildConfig(Arrays.copyOfRange(args, 1, args.length)));
+      default:
+        System.err.println("Unrecognized command '" + args[0] + "'");
+        System.exit(1);
+    }
+  }
+
+  private static int check(String[] args) throws IOException {
+    if (args.length == 1 && "--help".equals(args[0])) {
+      System.err.println(
+          "Analyze Temporal workflows for common mistakes.\n"
+              + "\n"
+              + "Usage:\n"
+              + "  workflowcheck check <classpath...> [--config <config-file>] [--no-default-config] [--show-valid]");
+      return 0;
+    }
+    // Args list that removes options as encountered
+    List<String> argsList = new ArrayList<>(Arrays.asList(args));
+
+    // Load config
+    List<Properties> configProps = new ArrayList<>();
+    if (!argsList.remove("--no-default-config")) {
+      configProps.add(Config.defaultProperties());
+    }
+    while (true) {
+      int configIndex = argsList.indexOf("--config");
+      if (configIndex == -1) {
+        break;
+      } else if (configIndex == argsList.size() - 1) {
+        System.err.println("Missing --config value");
+        return 1;
+      }
+      argsList.remove(configIndex);
+      Properties props = new Properties();
+      try (FileInputStream is = new FileInputStream(argsList.remove(configIndex))) {
+        props.load(is);
+      }
+      configProps.add(props);
+    }
+
+    // Whether we should also show valid
+    boolean showValid = argsList.remove("--show-valid");
+
+    // Ensure that we have at least one classpath arg
+    if (argsList.isEmpty()) {
+      System.err.println("At least one classpath argument required");
+      return 1;
+    }
+    // While it can rarely be possible for the first file in a class path string
+    // to start with a dash, we're going to assume it's an invalid argument and
+    // users can qualify if needed.
+    Optional<String> invalidArg = argsList.stream().filter(s -> s.startsWith("-")).findFirst();
+    if (invalidArg.isPresent()) {
+      System.err.println("Unrecognized argument: " + invalidArg);
+    }
+
+    System.err.println("Analyzing classpath for classes with workflow methods...");
```

Comment:
Not sure why I do `System.err` here and `System.out` below, probably just need to pick one